### ISSUE TITLE
cross entropy calculation mistake in logistic_regression.py line 36

### ISSUE THREAD

**Issue by iamharshit** [aymericdamien/TensorFlow-Examples#139]
In logistic_regression.py line 36, you are calculating the cross entropy loss as
```
cost = tf.reduce_mean(-tf.reduce_sum(y*tf.log(pred), reduction_indices=1))
```
but shouldn't instead of `reduce_mean` we been doing `reduce_sum` so
```
cost = tf.reduce_sum(-tf.reduce_sum(y*tf.log(pred), reduction_indices=1))
```
because the cross entropy formulae involves a summation sign in the begining.
Please correct me if i am wrong.

**Comment by TJCVRS** [comment]
@iamharshit You can visit the Ufldl  to check the cost formula, it's supposed to calculate the mean value of all the elements in training.

**Comment by coderbee** [comment]
The cross entropy formula also has a divide-by-m sign in addition to summation, so we use reduce_mean in the outermost expression.
